runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -3735,6 +3736,15 @@ struct amRequest_AMO_t {
   void* result;                 // result address on initiator's node
 };
 
+struct amRequest_FAMO_result_t {
+  struct amRequest_base_t b;
+  int8_t size;                  // object size (bytes)
```

Comment:
I don't think we need both the size and the type (`enum fi_datatype`). The type encodes the size, so we could just use the type. Alternatively, I believe that the offset of the value in a `chpl_amo_datum_t` is always zero, so we could also get by with just use the size.